vowpalwabbit/parse_args.cc

```diff
@@ -1202,16 +1193,94 @@ ssize_t trace_message_wrapper_adapter(void* context, const char* buffer, size_t
 VW::workspace& parse_args(
     std::unique_ptr<options_i, options_deleter_type> options, trace_message_t trace_listener, void* trace_context)
 {
-  VW::workspace& all = *(new VW::workspace());
+  auto logger = trace_listener != nullptr ? VW::io::create_custom_sink_logger(trace_context, trace_listener)
+                                          : VW::io::create_default_logger();
+
+  bool quiet = false;
+  bool driver_output_off = false;
+  std::string driver_output_stream;
+  std::string log_level;
+  std::string log_output_stream;
+  size_t upper_limit = 0;
+  option_group_definition logging_options("Logging options");
+  logging_options
+      .add(make_option("quiet", quiet)
+               .help("Don't output diagnostics and progress updates. Supplying this implies --log_level off and "
+                     "--driver_output_off."))
+      .add(make_option("driver_output_off", driver_output_off).help("Disable output for the driver."))
+      .add(make_option("driver_output_stream", driver_output_stream)
+               .default_value("stderr")
+               .one_of({"stdout", "stderr"})
+               .help("Specify the stream to output driver output to."))
+      .add(make_option("log_level", log_level)
+               .default_value("info")
+               .one_of({"info", "warn", "error", "critical", "off"})
+               .help("Log level for logging messages. Specifying this wil override --quiet for log output."))
+      .add(make_option("log_output_stream", log_output_stream)
+               .default_value("compat")
+               .one_of({"stdout", "stderr", "compat"})
+               .help("Specify the stream to output log messages to. In the past VW's choice of stream for logging "
+                     "messages wasn't consistent. Suppling compat will maintain that old behavior. Compat is now "
+                     "deprecated so it is recommended that stdout or stderr is chosen."))
+      .add(make_option("limit_output", upper_limit)
+               .default_value(0)
+               .help("Avoid chatty output. Limit total printed lines. 0 means unbounded."));
+
+  options->add_and_parse(logging_options);
+
+  if (quiet)
+  {
+    // If a user supplies an explicit log level, then we'll respect that.
```

Comment:
but specifying --quiet still overrides that log level right?

Reply:
As it is right now in the PR. Explicit log_level overrides quiet. Do you think quiet should always silence everything?

Reply:
Yeah i think quiet should have highest priority